D/ducc0/build_tarballs.jl

```diff
@@ -0,0 +1,65 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+using BinaryBuilderBase
+
+include(joinpath(@__DIR__, "..", "..", "platforms", "microarchitectures.jl"))
+
+name = "ducc0"
+version = v"0.27.0"
+
+# Collection of sources required to complete build
+sources = [
+GitSource("https://gitlab.mpcdf.mpg.de/mtr/ducc.git", "84967dd5d3e3062874a03c99a6d51ab375d3fb9d"),
+ArchiveSource("https://github.com/phracker/MacOSX-SDKs/releases/download/10.15/MacOSX10.15.sdk.tar.xz",
+                  "2408d07df7f324d3beea818585a6d990ba99587c218a3969f924dfcc4de93b62"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+
+if [[ "${target}" == x86_64-apple-darwin* ]]; then
+    # Install a newer SDK to work around compilation failures
+    pushd $WORKSPACE/srcdir/MacOSX10.*.sdk
+    rm -rf /opt/${target}/${target}/sys-root/System
+    cp -ra usr/* "/opt/${target}/${target}/sys-root/usr/."
+    cp -ra System "/opt/${target}/${target}/sys-root/."
+    popd
+    MACFLAGS=-mmacosx-version-min=10.14
+fi
+
+cd $WORKSPACE/srcdir/ducc*/julia
+install_license ../LICENSE
+${CXX} ${CFLAGS} ${MACFLAGS} -O3 -I ../src/ ducc_julia.cc -Wfatal-errors -pthread -std=c++17 -fPIC -fno-math-errno -fassociative-math -freciprocal-math -fno-signed-zeros -fno-trapping-math -ffp-contract=fast -ffinite-math-only -fno-rounding-math -fno-signaling-nans -fexcess-precision=fast -c
+# -fcx-limited-range is not supported by clang
+${CXX} ${CFLAGS} ${MACFLAGS} -O3 -o libducc_julia.${dlext} ducc_julia.o -Wfatal-errors -pthread -std=c++17 -shared -fPIC
+install -Dvm 0755 "libducc_julia.${dlext}" "${libdir}/libducc_julia.${dlext}"
+"""
+
+# Expand for microarchitectures on x86_64 (library doesn't have CPU dispatching)
+# Tests on Linux/x86_64 yielded a slow binary with avx512 for some reason, so disable that
+platforms = expand_cxxstring_abis(expand_microarchitectures(supported_platforms(), ["x86_64", "avx", "avx2"]); skip=!Sys.iswindows)
+
```

Comment:
```
┌ Warning: /agent/_work/1/s/D/ducc0/build/x86_64-linux-gnu-march+avx2/97WhZ0rQ/x86_64-linux-gnu-libgfortran5-cxx11-march+avx2/destdir/lib/libducc_julia.so contains std::string values!  This causes incompatibilities across the GCC 4/5 version boundary.  To remedy this, you must build a tarball for both GCC 4 and GCC 5.  To do this, immediately after your `platforms` definition in your `build_tarballs.jl` file, add the line:
│ 
│     platforms = expand_cxxstring_abis(platforms)
└ @ BinaryBuilder.Auditor /depot/packages/BinaryBuilder/7iMm2/src/auditor/compiler_abi.jl:247
```
```suggestion
platforms = expand_cxxstring_abis(platforms)

```

Reply:
Will this still produce separate packages for AVX and AVX2 capable targets? This is fairly important for the package.

Reply:
Yes.

Side note, support for microarchitectures is ***extremely*** experimental at the moment and liable to be changed in breaking way later.  This isn't even documented (or just barely) precisely for this reason.  More specifically, building the libraries works well, the problem is handling the installation of the artifacts, we haven't had the time to improve that.

Reply:
Thanks for pointing this out. I copied the block from https://github.com/JuliaPackaging/Yggdrasil/tree/master/F/finufft, since the requirements are pretty similar.

I'm happy to apply changes to this as and when they become necessary. If I may make a suggestion, please consider adding extra architectures for FMA3/4 support as well. Some numerical codes can get around 50% speedup when they are allowed to use these instructions.